03-MVC-AngularJS/js/script.js

```diff
@@ -0,0 +1,87 @@
+var myApp = angular.module('myApp', ['ngRoute']);
+
+myApp.config(function($routeProvider) {
+    $routeProvider
+    .when("/:id", {
+        template : (`<h1>Title: {{selected.title}}</h1><br>
+        <p>Duration: {{selected.duration}}</p>
+        <p>Year: {{selected.year}}</p>
+        <p>Synopsis: {{selected.synopsis}}</p>`)
+    })
+});
+
+myApp.service('storage',[function() {
+
+    this.editDisabled = true;
+
+    this.movies = JSON.parse(localStorage.getItem('movies'));
+
+    this.add = function (movieNew) {
+        this.movies.push(movieNew);
+    }
+
+    this.save  = function (moviesUnsaved) {
+        localStorage.setItem('movies', angular.toJson(moviesUnsaved));
+    }   
+
+    this.get  = function() {
+
+        if (this.movies === null) {
+            this.movies = [];
+        }
+        return this.movies;
+    }
+}]);
+
+
+myApp.controller('mainController', ['$scope', 'storage', function ($scope, storage) {
+    let moviesUpdate;
+
+    storage.showDetails = false;
+    $scope.movies = {};
+    $scope.selected = {};
+    moviesUpdate = storage.get();
+    $scope.movies.details = moviesUpdate;
+    $scope.showDetails = storage.showDetails;
+
+    $scope.$watch('movies.details', function(moviesUpdate) {storage.save(moviesUpdate);}, true);
```

Comment:
When you have every method to manipulate the data inside the service you will be able to remove this watcher just calling your service.save method after each add, update or delete. Your save method shoud looks something like:
```

this.save = function () {
    localStorage.setItem('movies', angular.toJson(this.movies));
};
```